components/src/core/Drawer/Drawer.tsx

```diff
@@ -283,7 +283,15 @@ export const DrawerComponent: React.FC<DrawerComponentProps> = (props) => {
                 drawerProps.style
             )}
         >
-            <div className={clsx(defaultClasses.content, classes.content)} style={{ width: contentWidth }}>
+            <div
+                className={clsx(
+                    defaultClasses.content,
+                    classes.content,
+                    'pxb-drawer-content',
+                    isDrawerOpen() && 'pxb-drawer-open'
```

Comment:
Rather than passing around these non-JSS class names, I would recommend adding a React Context around the drawer that makes some drawer context available to its various children (like the DrawerNavItem), mostly just the open state right now, but we could possibly use it for some of the prop inheritance logic too down the road.

If you look at the Header component in RN, we use a couple of contexts to pass information down this way (such as the height of the animated header): https://github.com/pxblue/react-native-component-library/blob/dev/components/src/core/header/header.tsx#L287

The context provider is defined here: https://github.com/pxblue/react-native-component-library/blob/dev/components/src/core/header/contexts/HeaderHeightContextProvider.tsx

You could then just conditionally add your noIconTitle class below based on:
```
[defaultClasses.noIconTitle]: hidePadding && !icon && !drawerOpen
```

When setting up the context, you'll want to make sure that if context === null, you'll just want to return a default value like true so that the UserMenu and DropdownToolbar still work as expected (they use the DrawerNavItem but won't have a Drawer parent to learn the open state from).

Here's some more reading on Context for background: https://reactjs.org/docs/context.html